tests/conftest.py

```diff
@@ -52,6 +52,21 @@
     from tests.types import ProjectFactory
 
 
+# Enable handling long path names (260) in windows
+# https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation
+if sys.platform.startswith("win32"):
+    import subprocess
+
+    git_command = find_git_command()
+    env = os.environ.copy()
+    env["GIT_TERMINAL_PROMPT"] = "0"
+    args = ["config", "--system", "core.longpaths", "true"]
```

Comment:
IIUC, we are changing a git setting globally? Can we do it locally (`--local`)?

Reply:
local doesn't work. I don't know how the test pipeline si setup. It took me a while to figure out it needs to be set more broadly. Alternatively, the option could be set somewhere where the test environment of CI. I'd need some help with that.

Reply:
it's not ok for running test scripts to have the side-effect of changing the user's global git configuration, though!

- can a way be found to make the paths shorter in the first place?
- if this has to be system configuration, it should probably be run from the CI workflow yaml definition - so that it only affects CI

Reply:
That's a very good point. I don't know why I was thinking only in the context of the CI.

Getting the path length just bellow the limit is just hiding it for the next guy on linux who won't know why its failing on windows. It's failing on a git subprocess, so the error is not super easy to spot, especially since the subprocess stderr is truncated.

I have come across an interesting fact; that Python for windows installer has an option to [Enable long paths on windows via a registry key when the installer is run](https://stackoverflow.com/questions/51019926/python-installer-for-windows-disable-path-length-limit-option-not-available). This leads me to believe that the complication is somewhat common and avoidable with that without drawbacks as the registry key help text doesn't mention any.

I will look into moving it to CI.

Reply:
I've put the long pathname config into CI.

I think that the reason --local doesn't work is because during tests, different repos are handled and --local sets it only for the current poetry repo which is not where the issue manifests itself.